server/doc.js

```diff
@@ -0,0 +1,37 @@
+const mongoose = require('mongoose');
+mongoose.promise = Promise;
+const Schema = mongoose.Schema;
+const date = require('date-and-time');
+
+const docSchema = new Schema({
+  name: {type: String, required = true},
+  file: {type: String, required = true},
+  owner_id: {type: Number, required = true},
+  locked: Boolean,
+  revision_id: Number,
+  last_revised: {type: Date, required = true}
+})
+
+docSchema.methods.filter = function(taboo_words){
```

Comment:
This method is not needed. We instead need to generate a list of indexes indicating the locations of all taboo words.

Reply:
changed to return a arr of starts and ends for all places where taboo words show up